### PR TITLE
Set initial node-pg-pool size through environment

### PR DESCRIPTION
**Pull request by fabubaker**
Introduces a new env var, `NODEPG_INITIAL_MIN_CLIENTS` that sets the inital size of the node-pg-pool.

The pool size is set before the server starts listening for requests.

### CODE REVIEW

File: server/src/adapters/postgres/postgres.js [hunk 1]
```diff
@@ -499,6 +499,20 @@ module.exports = {
     return pool.query('SELECT NOW() as now')
       .then(function (res) {
         console.log('Connected to postgres on', res.rows[0].now)
+
+        // Initialize pool with NODEPG_INITIAL_MIN_CLIENTS
+        var promises = _.times(parseInt(process.env.NODEPG_INITIAL_MIN_CLIENTS || '10'), () => {
```
Comment (hammady) [inline server/src/adapters/postgres/postgres.js]: With this `times`, there would be an additional client for the query in line 499, I stand corrected.

File: server/src/adapters/postgres/postgres.js [hunk 2]
```diff
@@ -499,6 +499,20 @@ module.exports = {
...
+          return pool
+            .connect()
+            .then((connection) => {
+              return connection.release()
+            })
+            .catch((error) => {
+              console.log('Error in initializing pool: ', error)
```
Comment (hammady) [inline server/src/adapters/postgres/postgres.js]: Are we OK with silencing errors in the initial pool warm-up?

Comment (fabubaker) [reply server/src/adapters/postgres/postgres.js]: You're right, I've removed this `catch` block so that the error is handled in `app.js` instead.

Comment (hammady) [reply server/src/adapters/postgres/postgres.js]: Where is that change?

File: server/src/adapters/postgres/postgres.js (hunk 1 above)
Comment (fabubaker) [reply server/src/adapters/postgres/postgres.js]: The pool has a client in the beginning and this is used in `pool.query('SELECT NOW() as now')`, so there is no extra client created.

**Comment by fabubaker** [comment]
> This implementation does not guarantee a minimum that is always satisfied.

Right, this is with the assumption that `NODEPG_CLIENT_TIMEOUT` is set to 0.

File: server/src/adapters/postgres/postgres.js (hunk 2 above)
Comment (fabubaker) [reply server/src/adapters/postgres/postgres.js]: https://github.com/rayyanqcri/rayyan-store/pull/90/commits/fba4f9504b5cf9d8d3fa1a5631540cc3a0f2d3d9

**Pull request merged**